homeassistant/components/tado/binary_sensor.py

```diff
@@ -125,3 +168,107 @@ def _async_update_device_data(self):
             self._state = self._device_info.get("connectionState", {}).get(
                 "value", False
             )
+
+
+class TadoZoneBinarySensor(TadoZoneEntity, BinarySensorEntity):
+    """Representation of a tado Sensor."""
+
+    def __init__(self, tado, zone_name, zone_id, zone_variable):
+        """Initialize of the Tado Sensor."""
+        self._tado = tado
+        super().__init__(zone_name, tado.home_id, zone_id)
+
+        self.zone_variable = zone_variable
+
+        self._unique_id = f"{zone_variable} {zone_id} {tado.home_id}"
+
+        self._state = None
+        self._state_attributes = None
+        self._tado_zone_data = None
+
+    async def async_added_to_hass(self):
+        """Register for sensor updates."""
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                SIGNAL_TADO_UPDATE_RECEIVED.format(
+                    self._tado.home_id, "zone", self.zone_id
+                ),
+                self._async_update_callback,
+            )
+        )
+        self._async_update_zone_data()
+
+    @property
+    def unique_id(self):
+        """Return the unique id."""
+        return self._unique_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self.zone_name} {self.zone_variable}"
+
+    @property
+    def is_on(self):
+        """Return true if sensor is on."""
+        return self._state
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        if self.zone_variable == "early start":
+            return DEVICE_CLASS_POWER
+        if self.zone_variable == "link":
+            return DEVICE_CLASS_CONNECTIVITY
+        if self.zone_variable == "open window":
+            return DEVICE_CLASS_WINDOW
+        if self.zone_variable == "overlay":
+            return DEVICE_CLASS_POWER
+        if self.zone_variable == "power":
+            return DEVICE_CLASS_POWER
+        return None
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._state_attributes
+
+    @callback
+    def _async_update_callback(self):
+        """Update and write state."""
+        self._async_update_zone_data()
+        self.async_write_ha_state()
+
+    @callback
+    def _async_update_zone_data(self):
+        """Handle update callbacks."""
+        try:
+            self._tado_zone_data = self._tado.data["zone"][self.zone_id]
+        except KeyError:
+            return
+
+        if self.zone_variable == "power":
+            self._state = self._tado_zone_data.power
+
+        elif self.zone_variable == "link":
+            self._state = self._tado_zone_data.link
+
+        elif self.zone_variable == "overlay":
+            self._state = self._tado_zone_data.overlay_active
+            self._state_attributes = (
+                {"termination": self._tado_zone_data.overlay_termination_type}
```

Comment:
What does the termination attribute measure?

Reply:
I just copied this from the non binary sensor and moved it to the binary sensor,.
I think `termination` indicates when the overlay mode is going to end. This only applies if the user configured an overlay until the next scheduled change:
```json
    "overlay": {
        "setting": {
            "power": "ON",
            "temperature": {
                "celsius": 20.0,
                "fahrenheit": 68.0
            },
            "type": "HEATING"
        },
        "termination": {
            "projectedExpiry": null,
            "type": "TADO_MODE",
            "typeSkillBasedApp": "TADO_MODE"
        },
        "type": "MANUAL"
    },
```

Reply:
We need to know what values we're setting as attributes to know if it's following our design standards. Timestamps eg should be absolute UTC isormatted and normally exposed as a separate sensor entity with device class timestamp.

Reply:
I've removed _state_attributes from binary sensors (these attributes aren't really useful).